packages/main/src/plugin/extension-loader.ts

```diff
@@ -1320,10 +1320,19 @@ export class ExtensionLoader {
         if (options.images) {
           options.images.icon = instance.updateImage(options?.images?.icon, extensionPath);
         }
-        const cliTool = this.cliToolRegistry.createCliTool(extensionInfo, options);
+        const cliTool = instance.cliToolRegistry.createCliTool(extensionInfo, options);
         disposables.push(cliTool);
         return cliTool;
       },
+      getCliTool: (id: string): containerDesktopAPI.CliTool | undefined => {
+        return instance.cliToolRegistry.getCliTool(id);
+      },
+      get all() {
+        return instance.cliToolRegistry.getCliTools();
```

Comment:
the object that we're returning has still all the properties, it should be a subset (new object) as here it's possible to call all the methods